Support more Qualcomm SoCs

Support more Qualcomm SoCs when use Qualcomm AI Engine Direc.

@chiwwang I can't connect to `SA8295` through `adb` directly.  So i connected to `8 Gen1 (V69)` through `adb`  and run `adb  shell "./qnn_executor_runner"` to do some test.
I found a strange issue. 
If i built with `- m SA8295`, it works.
```
python -m examples.qualcomm.scripts.deeplab_v3 -b build_android -m SA8295
```
if i built with `-m SM8550`, it resulted in the following error.
```
[ERROR] <E> Unsupported HTP arch 73 from device custom config
```
So i think `PopulateSocInfoTable` didn't run as expected."

@chiwwang 
```
# adb shell " ls -l /data/local/tmp/executorch/dlv3_qnn|grep Htp"
-rwxrwxrwx 1 shell shell   1257600 2023-11-07 15:57 libQnnHtp.so
-rwxrwxrwx 1 shell shell   6057216 2023-11-07 15:57 libQnnHtpV68Skel.so
-rwxrwxrwx 1 shell shell    376616 2023-11-07 15:57 libQnnHtpV68Stub.so

# adb shell "./qnn_executor_runner --model_path ./dlv3_qnn.pte"
[WARNING] <W> Initializing HtpProvider
[ERROR] <E> Failed in loading stub: dlopen failed: library "libQnnHtpV69Stub.so" not found
[ERROR] <E> Unable to load Remote symbols 4000
[ERROR] <E> Unable to load Remote symbols 4000
```
I think the error should be as following if i build with `-m SA8295` and run it on `8 Gen1`.
```
[ERROR] <E> Unsupported HTP arch 68 from device custom config
```

> I think the error should be as following if i build with -m SA8295 and run it on 8 Gen1.
> [ERROR] <E> Unsupported HTP arch 68 from device custom config

Strictly speaking, yes. We cannot assume V68 binary work on V69 devices. But the error might a bit different from the above.


So did you try the below command on `8 Gen 1`?
```
python -m examples.qualcomm.scripts.deeplab_v3 -b build_android -m SM8450
```
I think it should work, and the script should push correct `libQnnHtpV69Stub.so`.

As an overall picture, the flow is
1. The script `deeplab_v3.py` compile the model against a specific SoC, passed by `-m` argument.
2. It also push required `.so` for the specific SoC. For example, `libQnnHtpV69Stub/Skel.so` for `8 Gen 1`
In `SA8295` case, as you added in `examples/qualcomm/..../utils.py`, it's `libQnnHtpV68Stub/Skel.so`
3. Then when we run `./qnn_executor_runner` on a device, it performs:
a. dlopen `libQnnHtp.so`
b. `libQnnHtp.so` check the device spec, and open corresponding `libQnnHtpVxxStub.so`
c. `libQnnHtpVxxStub.so` requests Hexagon to open `libQnnHtpVxxSkel.so`.

So you see the error message `[ERROR] <E> Failed in loading stub: dlopen failed: library "libQnnHtpV69Stub.so" not found`
because of the above step `b.`

I build my code use this command "python -m examples.qualcomm.scripts.deeplab_v3 -b build_android -m SM8550 --compile_only --download", run on my SA8155 chipset, and get the following error:
hx11:/data/local/tmp/executorch_qualcomm_tutorial # ./qnn_executor_runner --model_path ./dlv3_qnn.pte
I 00:00:00.037259 executorch:qnn_executor_runner.cpp:81] Model file ./dlv3_qnn.pte is loaded.
I 00:00:00.037308 executorch:qnn_executor_runner.cpp:90] Using method forward
I 00:00:00.037320 executorch:qnn_executor_runner.cpp:138] Setting up planned buffer 0, size 9031680.
[INFO][Qnn ExecuTorch] create QNN Logger with log_level 2
[WARNING] <W> Initializing HtpProvider

[ERROR] <E> Unsupported SnapdragonModel by HTP backend

[ERROR] <E> Init support device platform info failed

[ERROR] <E> Failed to init router

[ERROR] <E> QnnLog_create - Provider initialization never called

[WARNING][Qnn ExecuTorch] Failed to create log_handle for backend  6, qnn_log_level 2, error=11002
[INFO][Qnn ExecuTorch] Initialize Qnn backend parameters for Qnn executorch backend type 2
[INFO][Qnn ExecuTorch] Caching: Caching is in RESTORE MODE.
[WARNING] <W> Initializing HtpProvider

[ERROR] <E> Unsupported SnapdragonModel by HTP backend

[ERROR] <E> Init support device platform info failed

[ERROR] <E> Failed to init router

[ERROR] <E> Failed to initialize provider

[ERROR][Qnn ExecuTorch] Failed to create backend_handle for Backend ID 6, error=4000
E 00:00:00.046192 executorch:QnnManager.cpp:84] Fail to configure Qnn backend
E 00:00:00.046199 executorch:QnnExecuTorchBackend.cpp:143] Fail to initialize Qnn Manager
E 00:00:00.046204 executorch:method.cpp:106] Init failed for backend QnnBackend: 0x1
F 00:00:00.046211 executorch:qnn_executor_runner.cpp:160] In function main(), assert failed (method.ok()): Loading of method forward failed with status 0x1.
I want to add SA8155 into executorch and I'm wondering if there's anything I need to pay attention to, what version of HTP architecture should I use? Has anyone done this before?

--- a/backends/qualcomm/serialization/qc_compiler_spec.fbs
+++ b/backends/qualcomm/serialization/qc_compiler_spec.fbs
@@ -36,6 +36,7 @@ enum QcomChipset: int {
   SM8550 = 43,
   SSG2115P = 46,
   SM8650 = 57,
+  SA8295 = 39,
 }
 
 /// Indicate the information of the specified SoC.

--- a/backends/qualcomm/serialization/qc_schema.py
+++ b/backends/qualcomm/serialization/qc_schema.py
@@ -41,6 +41,7 @@ class QcomChipset(IntEnum):
     SM8550 = 43  # v73
     SSG2115P = 46  # v73
     SM8650 = 57  # v75
+    SA8295 = 39  # v68
 
 
 @dataclass
@@ -55,6 +56,7 @@ _soc_info_table = {
     QcomChipset.SM8550: SocInfo(QcomChipset.SM8550, HtpInfo(HtpArch.V73, 8)),
     QcomChipset.SM8650: SocInfo(QcomChipset.SM8650, HtpInfo(HtpArch.V75, 8)),
     QcomChipset.SSG2115P: SocInfo(QcomChipset.SSG2115P, HtpInfo(HtpArch.V73, 2)),
+    QcomChipset.SA8295: SocInfo(QcomChipset.SA8295, HtpInfo(HtpArch.V68, 8)),
 }
 
 

--- a/backends/qualcomm/utils/utils.py
+++ b/backends/qualcomm/utils/utils.py
@@ -967,6 +967,7 @@ def get_soc_to_arch_map():
         "SM8550": HtpArch.V73,
         "SM8475": HtpArch.V69,
         "SM8450": HtpArch.V69,
+        "SA8295": HtpArch.V68,
     }
 
 
@@ -977,6 +978,7 @@ def get_soc_to_chipset_map():
         "SM8550": QcomChipset.SM8550,
         "SM8475": QcomChipset.SM8475,
         "SM8450": QcomChipset.SM8450,
+        "SA8295": QcomChipset.SA8295,
     }
 
 

